### PR TITLE
Tests now log to a file by default

### PR DESCRIPTION
**Pull request by etorrejon**
Logs are limited in size and number.

Resolves https://github.com/actionhero/actionhero/issues/1226

### CODE REVIEW

**Comment by evantahler** [comment]
Thank you for this work!  I like your choice to limit the test log file size.
You have a lot of work here to generate the log directory... but that's not needed! Winston will do that for us.

File: config/logger.js [hunk 1]
```diff
@@ -3,6 +3,16 @@
 const fs = require('fs')
 const cluster = require('cluster')
 
+function createDirectory (path) {
```
Comment (evantahler) [inline config/logger.js]: Winston will do this (and exit if it can't) for us

File: config/logger.js [hunk 2]
```diff
@@ -19,24 +29,21 @@ exports['default'] = {
     }
 
     // file logger
-    logger.transports.push(function (api, winston) {
-      if (api.config.general.paths.log.length === 1) {
+    const hasLogDirectoryConfigured = (api.config.general.paths.log.length === 1)
```
Comment (evantahler) [inline config/logger.js]: I like wrapping the path check into a method!

File: config/logger.js [hunk 3]
```diff
@@ -53,8 +60,27 @@ exports['default'] = {
...
+
+    if (hasLogDirectoryConfigured) {
+      logger.transports.push(function (api, winston) {
+        const logDirectory = api.config.general.paths.log[0]
+
+        createDirectory(logDirectory)
+
+        return new (winston.transports.File)({
+          filename: api.config.general.paths.log[0] + '/' + api.pids.title + '.log',
+          maxsize: 20480,
```
Comment (evantahler) [inline config/logger.js]: Great idea!

File: config/logger.js (hunk 2 above)
Comment (etorrejon) [reply config/logger.js]: thanks! :)

File: config/logger.js (hunk 3 above)
Comment (etorrejon) [reply config/logger.js]: :)

**Comment by etorrejon** [comment]
Thanks for the review, @evantahler !! I'll update

**Comment by evantahler** [comment]
Thanks! Let us know if you need any help.

**Comment by evantahler** [comment]
Fascinating! It looks like Winston will create any log files it needs... but not the directory if it doesn't exist.  

So, do you A) create the directory within the initializer or B) create the directory as part of the test setup (https://github.com/actionhero/actionhero/blob/master/.circleci/config.yml#L13). ?

**Comment by etorrejon** [comment]
Creating in the initializer feels like a good place for it - I think that way it'll create the dir even if you're not running tests...

If that works for you, I'll add a commit that does that.

**Comment by etorrejon** [comment]
Actually I guess it'd be in `config/logger.js` since that's where the transport is configured

**Comment by etorrejon** [comment]
... unless you're thinking it'd be cleaner to leave that code out of config by keeping it in the build script

**Comment by etorrejon** [comment]
ok sounds good! Sorry for the delay - will have a PR up soon.

**Comment by evantahler** [comment]
Thansk!

**Pull request merged**